### ISSUE TITLE
cannot load such file -- active_support/dependencies from load_path_cache

### ISSUE THREAD

**Issue by frederikspang** [Shopify/bootsnap#180]
Hi,

First of all, thanks for a great project, that have decreased our loading time from ~60 to 8 seconds, in a huge project..

However, I am running into an issue, when running `$ rails s` instead of `bundle exec rails s`, that throws an error... I have run out of ideas as to fixing it; I have cleared the `RAILS_ROOT/tmp/cache/bootsnap_*`-caches, and been looking for answers in google for a few days..

**STACK TRACE (sanitized paths)**

```
~/rvm_path/gems/ruby-2.3.3@gemset/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require': cannot load such file -- active_support/dependencies (LoadError)
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache.rb:39:in `setup'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/bootsnap-1.3.0/lib/bootsnap.rb:24:in `setup'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/bootsnap-1.3.0/lib/bootsnap/setup.rb:33:in `<top (required)>'
  from ~/rvm_path/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:133:in `require'
  from ~/rvm_path/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:133:in `rescue in require'
  from ~/rvm_path/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:40:in `require'
  from ~/rails_root/config/boot.rb:19:in `<top (required)>'
  from ~/rails_root/bin/rails:8:in `require_relative'
  from ~/rails_root/bin/rails:8:in `<top (required)>'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/spring-2.0.2/lib/spring/client/rails.rb:28:in `load'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/spring-2.0.2/lib/spring/client/rails.rb:28:in `call'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/spring-2.0.2/lib/spring/client/command.rb:7:in `call'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/spring-2.0.2/lib/spring/client.rb:30:in `run'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/spring-2.0.2/bin/spring:49:in `<top (required)>'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/spring-2.0.2/lib/spring/binstub.rb:31:in `load'
  from ~/rvm_path/gems/ruby-2.3.3@gemset/gems/spring-2.0.2/lib/spring/binstub.rb:31:in `<top (required)>'
  from ~/rvm_path/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
  from ~/rvm_path/rubies/ruby-2.3.3/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `require'
  from ~/rails_root/bin/spring:15:in `<top (required)>'
  from bin/rails:3:in `load'
  from bin/rails:3:in `<main>'
```

**ENVIRONMENT**
 * Rails 4.2.7 (tried on 4.2.10 as well).
 * Ruby 2.3.3
 * Bootsnap locked to 1.3.0 in Gemfile.lock

**Expected behaviour **
Rails server boots as usual, as when running `bundle exec rails s`

**Actual behaviour**
Both `rails s` and `bin/rails s` throw error as seen above, and fail.

**Other notes**
Clearing bootsnap cache doesnt work, nor does running `spring stop` beforehand.

**Comment by iChip** [comment]
Having same issue, don't use Spring in my project.

**Comment by iChip** [comment]
https://github.com/Shopify/bootsnap/blob/master/lib/bootsnap/load_path_cache.rb#L39

Possibly relates to https://github.com/rails/rails/issues/24818

**Comment by rafaelfranca** [comment]
That is a different issue. I think the the problem here is that the gem active support is not yet when `bootsnap/setup` is required. Can you try to add 'gem 'activesupport'` before the require of bootsnap?

**Comment by iChip** [comment]
Just tried adding `gem 'activesupport'` above `gem 'bootsnap', require: false` in Gemfile

Also tried with and without adding  `require 'active_support'` above `require 'bootsnap/setup'` in config/boot.rb

No change

**Comment by rafaelfranca** [comment]
that is strange. It should be able to load active_support. Are you able to reproduce this in a new application?

**Comment by dgarwood** [comment]
@frederikspang I'm curious if you've tried to remove bootsnap from your project and replicate the problem.  I'm working through a similar issue (which lead me to this issue), but when I turned off bootsnap, the problem was still there.  it turns out I was having an issue loading the dependency in active_support.

**Comment by neontuna** [comment]
I ran into this issue after upgrading from ruby 2.3.1 to ruby 2.4.4.  On top of what was suggested here, I tried a lot of (probably unnecessary) fixes. Including `gem pristine --all --only-executables` and `bundle install --force`  

I also tried a rails new with basically the same Gemfile as the problem project and didn't have any issues.

Finally I wiped out the entire /tmp/ directory from my project and that did it.
